packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -178,32 +179,30 @@ export class Rule extends Resource implements IRule {
       const sourceAccount = sourceStack.account;
       const sourceRegion = sourceStack.region;
 
-      if (targetRegion !== sourceRegion) {
```

Comment:
I can't click on it, and not your foul-up, but the code above this is actually wrong.

```ts
      const targetAccount = targetStack.account;
      const targetRegion = targetStack.region;
```

Needs to change into:

```ts
      const targetAccount = targetProps.targetResource.env.account;
      const targetRegion = targetProps.targetResource.env.region;
```

And honestly probably the same for sourceAccount etc.

Reply:
From the docs:

Reply:
Yes for created resources, no for imported resources.